src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/GeneratedFile.cs

```diff
@@ -0,0 +1,65 @@
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.MnemonicDB.Attributes.Extensions;
+using NexusMods.MnemonicDB.Abstractions;
+using File = NexusMods.Abstractions.Loadouts.Files.File;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+public static class GeneratedFile
+{
+    private const string Namespace = "NexusMods.Abstractions.Loadouts.Files.GeneratedFile";
+    
+    /// <summary>
+    /// The generator unique ID for the file, this specifies the class that is used to generate
+    /// the contents of this file. 
+    /// </summary>
+    public static readonly ClassAttribute<IFileGenerator> Generator = new("NexusMods.Abstractions.Loadouts.Files.GeneratedFile", nameof(Generator)) { IsIndexed = true };
+
+    public class Model(ITransaction tx) : File.Model(tx)
+    {
+        /// <summary>
+        /// Get an instance of the generator for this file.
+        /// </summary>
+        public IFileGenerator Generator
+        {
+            get => GeneratedFile.Generator.GetInstance(this);
+        }
+        
+        /// <summary>
+        /// Set the generator for this file.
+        /// </summary>
+        public void SetGenerator<TType>()
+        where TType : IFileGenerator
+        {
+            GeneratedFile.Generator.Add(this, TType.Guid);
+        }
```

Comment:
This looks super weird, why isn't this one property with a getter and setter like all the other models?

Reply:
Because you can't provide a type parameter to a property. It's really dumb but what I want is TType.Guid...and there's no way to do that in a property, because `Value` in this case would be IFileGenerator. Maybe there's a way to fix this, because yes it is ugly.